lib/vm/src/instance/mod.rs

```diff
@@ -1015,7 +1018,7 @@ impl InstanceHandle {
     /// Only safe to call immediately after instantiation.
     pub unsafe fn finish_instantiation(
         &self,
-        trap_handler: &dyn TrapHandler,
+        trap_handler: Arc<Option<Box<TrapHandler<'static>>>>,
```

Comment:
Generally for any of these functions, but you can make them take`&Arc`, too,  especially if they're not storing the `Arc` (in the cases where they just want to read it, you can even pass `Option<&TrapHandler<'static>>` I think by deref'ing the `Arc` and using `as_ref` on the `Option`)